### PR TITLE
Add basic support for profiling with Caliper

### LINKED ISSUE #119: Add option to use Caliper performance analysis library
We should have a build option to use [Caliper](https://github.com/LLNL/Caliper) for performance analysis and timing metrics.

### PR DESCRIPTION
**Pull request by joshessman-llnl**
This introduces the build system changes required to install and link against Caliper (with the `+caliper`) variant, documentation for source code annotation, and a simple example usage in a fairly hot loop within one of the integrators.

The example has no overhead when profiling is disabled, but it can be placed elsewhere if that would be more helpful.

Resolves #119

### CODE REVIEW

File: src/common/profiling.hpp [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+#include <string>
+
+#include "serac_config.hpp"
+
+#ifdef SERAC_USE_CALIPER
+
+#include "caliper/cali-manager.h"
+#include "caliper/cali.h"
+
+#define MARK_FUNCTION CALI_CXX_MARK_FUNCTION
```
Comment (white238) [inline src/common/profiling.hpp]: Paranoid Chris: Please prefix these with SERAC_ so we avoid name conflicts

**Review by white238 — APPROVED** [review]

**Comment by white238** [comment]
LGTM

File: src/drivers/serac.cpp [hunk 2]
```diff
@@ -29,6 +30,9 @@ int main(int argc, char* argv[])
 {
   auto [num_procs, rank] = serac::initialize(argc, argv);
 
+  // Initialize profiling if enabled
```
Comment (jamiebramwell) [inline src/drivers/serac.cpp]: Should this be in initialize?

Comment (joshessman-llnl) [reply src/drivers/serac.cpp]: Oops, I added it to initialize but forgot to remove it from the driver

**Review by jamiebramwell — APPROVED** [review]

**Comment by joshessman-llnl** [comment]
LGTM

**Comment by joshessman-llnl** [comment]
LGTM

**Pull request merged**